### DISCUSSION TITLE
Alpine build with WSL

### DISCUSSION THREAD

**Discussion by co-pas** [citizenfx/fivem#2703] [Engineering Discussion]
I'm trying to generate the server code using the resources provided by the project, 

fivem\code\tools\fxd> .\wsl-build.ps1

the WSL is even generated but when it comes to generating the build in the context created, I have problems. Is this currently possible in the project?

**Comment by co-pas** [comment]
error /mnt/c/fivem/ext/native-doc-tooling/node_modules/ffi-napi: Command failed.
Exit code: 1
Command: node-gyp-build
Arguments:
Directory: /mnt/c/fivem/ext/native-doc-tooling/node_modules/ffi-napi
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@5.1.0
gyp info using node@12.22.12 | linux | x64
gyp info find Python using Python version 3.12.4 found at "/tmp/py-venv/bin/python"
gyp http GET https://unofficial-builds.nodejs.org/download/release/v12.22.12/node-v12.22.12-headers.tar.gz
gyp http 200 https://unofficial-builds.nodejs.org/download/release/v12.22.12/node-v12.22.12-headers.tar.gz
gyp http GET https://unofficial-builds.nodejs.org/download/release/v12.22.12/SHASUMS256.txt
gyp http 200 https://unofficial-builds.nodejs.org/download/release/v12.22.12/SHASUMS256.txt
gyp info spawn /tmp/py-venv/bin/python
gyp info spawn args [
gyp info spawn args   '/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/mnt/c/fivem/ext/native-doc-tooling/node_modules/ffi-napi/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/root/.cache/node-gyp/12.22.12/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/root/.cache/node-gyp/12.22.12',
gyp info spawn args   '-Dnode_gyp_dir=/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/root/.cache/node-gyp/12.22.12/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/mnt/c/fivem/ext/native-doc-tooling/node_modules/ffi-napi',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
Traceback (most recent call last):
  File "/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py", line 50, in <module>
    sys.exit(gyp.script_main())
             ^^^^^^^^^^^^^^^^^
  File "/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 554, in script_main
    return main(sys.argv[1:])
           ^^^^^^^^^^^^^^^^^^
  File "/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 547, in main
    return gyp_main(args)
           ^^^^^^^^^^^^^^
  File "/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 520, in gyp_main
    [generator, flat_list, targets, data] = Load(
                                            ^^^^^
  File "/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 136, in Load
    result = gyp.input.Load(build_files, default_variables, includes[:],
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 2782, in Load
    LoadTargetBuildFile(build_file, data, aux_data,
  File "/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 391, in LoadTargetBuildFile
    build_file_data = LoadOneBuildFile(build_file_path, data, aux_data,
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 234, in LoadOneBuildFile
    build_file_contents = open(build_file_path, 'rU').read()
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid mode: 'rU' while trying to load binding.gyp
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)
gyp ERR! System Linux 5.15.153.1-microsoft-standard-WSL2
gyp ERR! command "/tmp/node/node/bin/node" "/tmp/node/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /mnt/c/fivem/ext/native-doc-tooling/node_modules/ffi-napi
gyp ERR! node -v v12.22.12
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok

Test:/mnt/c/fivem/code/tools/ci#

**Comment by co-pas** [comment]
node-gyp-build's ffi-napi has been my problem.

**Reply by kerneljay** [reply]
Me too - any solution would be **GREATLY** appreciated!

**Comment by duy56028a7d-debug** [comment]
use 14 version nodejs
